src/BinSkim.Rules/PERules/BA2006.BuildWithSecureTools.cs

```diff
@@ -66,6 +66,29 @@ public IEnumerable<IOption> GetOptions()
             new PerLanguageOption<AdvancedMitigations>(
                 AnalyzerName, nameof(AdvancedMitigationsEnforced), defaultValue: () => AdvancedMitigations.None);
 
+        public override void Initialize(BinaryAnalyzerContext context)
```

Comment:
>Initialize [](http://example.com/codeflow?start=29&length=10)

This is a very clever fix, Shaopeng, and we could consider taking it. But there's not a test or release note with your proposal.

I strongly encourage the team to please always produce release notes and tests *before* diving into code.

I would have merged this change if those things had been available. :)

I will continue to repeat these points to the team over and over, that is my role. 

Authoring a release note is a trivial task. Now, authoring a comprehensive set of unit tests might be more expense than a draft PR warrants. In *all* cases, however, if you haven't investigated the existing test strategy, that's a gap.

Reply:
@eddynaka or @shaopeng-gh, if you see these, please add the following PR note to this change, which I would like to merge. If someone has time to add a unit test as part of that, do so, otherwise please open an issue and we'll add a test as an offline activity.

## BUGFIX: Fix `KeyNotFoundException` on `BA2006.BuildWithSecureTools` when required properties are removed from the config file. [#565](https://github.com/microsoft/binskim/pull/565)

Reply:
sounds good.